addons/web_editor/controllers/main.py

```diff
@@ -542,11 +548,22 @@ def _load_resources(self, file_type, views, bundles_restriction, only_user_custo
         return files_data_by_bundle
 
     @http.route('/web_editor/modify_image/<model("ir.attachment"):attachment>', type="json", auth="user", website=True)
-    def modify_image(self, attachment, res_model=None, res_id=None, name=None, data=None, original_id=None, mimetype=None, alt_data=None):
+    def modify_image(self, attachment, res_model=None, res_id=None, name=None, data=None, mimetype=None, alt_data=None):
         """
-        Creates a modified copy of an attachment and returns its image_src to be
-        inserted into the DOM.
+        Creates a modified copy of an attachment.
+
+        return: - The image_src of the new attachment to be inserted into the
+                DOM.
+                - The id of the original attachment (potentially created here if
+                the image comes from the backend).
         """
+        if attachment.res_field:
+            # An original attachment has to be created first. We are in a
```

Comment:
Note: before this commit, this was not needed as it was impossible to modify an image that came from the backend; the user had to first upload an image through the media dialog before being able to modify it.